src/orion/algo/ax/axoptimizer.py

```diff
@@ -45,13 +46,17 @@ def __init__(self,
             choose_generation_strategy_kwargs=choose_generation_strategy_kwargs)
 
         self._client_state = None
+        self._trials_map = {} # tmp
 
     @property
     def state_dict(self):
         """Return a state dict that can be used to reset the state of the algorithm."""
         state_dict = super(AxOptimizer, self).state_dict
+
         # AxClient.to_json_snapshot() is more like the current internal state of the client than independent a snapshot
         state_dict["_client_state"] = copy.deepcopy(self._client_state)
+        state_dict["_trials_map"] = self._trials_map
```

Comment:
We should `deepcopy` the `self._trials_map` to prevent updates of the `state_dict` after it leaving this function.
```suggestion
        state_dict["_trials_map"] = copy.deepcopy(self._trials_map)
```